src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -0,0 +1,325 @@
+            ParticipationService participationService, UserService userService, StudentParticipationRepository studentParticipationRepository,
+            SimpMessageSendingOperations messagingTemplate, ResultService resultService) {
+        super(submissionRepository, userService);
+        this.fileUploadSubmissionRepository = fileUploadSubmissionRepository;
+        this.resultRepository = resultRepository;
+        this.participationService = participationService;
+        this.studentParticipationRepository = studentParticipationRepository;
+        this.messagingTemplate = messagingTemplate;
+        this.resultService = resultService;
+    }
+
+    /**
+     * Handles file upload submissions sent from the client and saves them in the database.
+     *
+     * @param fileUploadSubmission the file upload submission that should be saved
+     * @param fileUploadExercise   the corresponding file upload exercise
+     * @param principal            the user principal
+     * @return the saved file upload submission
+     */
+    @Transactional
+    public FileUploadSubmission handleFileUploadSubmission(FileUploadSubmission fileUploadSubmission, FileUploadExercise fileUploadExercise, Principal principal) {
+        Optional<StudentParticipation> optionalParticipation = participationService.findOneByExerciseIdAndStudentLoginAnyState(fileUploadExercise.getId(), principal.getName());
+        if (!optionalParticipation.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.FAILED_DEPENDENCY, "No participation found for " + principal.getName() + " in exercise " + fileUploadExercise.getId());
+        }
+        StudentParticipation participation = optionalParticipation.get();
+
+        if (participation.getInitializationState() == InitializationState.FINISHED) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "You cannot submit more than once");
+        }
+
+        return save(fileUploadSubmission, participation);
+    }
+
+    /**
+     * Given an exerciseId, returns all the file upload submissions for that exercise, including their results. Submissions can be filtered to include only already submitted
+     * submissions
+     *
+     * @param exerciseId    - the id of the exercise we are interested into
+     * @param submittedOnly - if true, it returns only submission with submitted flag set to true
+     * @return a list of file upload submissions for the given exercise id
+     */
+    @Transactional(readOnly = true)
+    public List<FileUploadSubmission> getFileUploadSubmissions(Long exerciseId, boolean submittedOnly) {
+        List<StudentParticipation> participations = studentParticipationRepository.findAllByExerciseIdWithEagerSubmissionsAndEagerResultsAndEagerAssessor(exerciseId);
+        List<FileUploadSubmission> submissions = new ArrayList<>();
+        for (StudentParticipation participation : participations) {
```

Comment:
This is a good suggestion, but it doesn't work for me. I didn't use streams in Java much, so maybe if you can help me and correct the suggestion, I can implement that. Otherwise, let's leave it as it is for now

Reply:
There's a .stram() missing in the first line. Have you tried it after adding this?

Reply:
Yes, I definetely missed it, now it is added. Thanks for helping!